src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs

```diff
@@ -294,31 +299,40 @@ private AnalyzerConfigDocument FindOrGenerateEditorConfig()
                 }
             }
 
-            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2)>.Empty;
+            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2, bool)>.Empty;
         }
 
-        private SourceText GetNewAnalyzerConfigDocumentText(
-            SourceText originalText,
-            Dictionary<string, TextLine> headers)
+        private SourceText GetNewAnalyzerConfigDocumentText(SourceText originalText)
         {
             // Check if an entry to configure the rule severity already exists in the .editorconfig file.
             // If it does, we update the existing entry with the new severity.
-            var configureExistingRuleText = CheckIfRuleExistsAndReplaceInFile(originalText, headers);
-            if (configureExistingRuleText != null)
+            var configureExistingRuleTextAndValidHeader = CheckIfRuleExistsAndReplaceInFile(originalText);
+            if (configureExistingRuleTextAndValidHeader.Item1 != null)
             {
-                return configureExistingRuleText;
+                return configureExistingRuleTextAndValidHeader.Item1;
             }
 
             // We did not find any existing entry in the in the .editorconfig file to configure rule severity.
             // So we add a new configuration entry to the .editorconfig file.
-            return AddMissingRule(originalText, headers);
+            return AddMissingRule(originalText, configureExistingRuleTextAndValidHeader.Item2, configureExistingRuleTextAndValidHeader.Item3);
         }
 
-        private SourceText CheckIfRuleExistsAndReplaceInFile(
-            SourceText result,
-            Dictionary<string, TextLine> headers)
+        private Tuple<SourceText, TextLine?, TextLine?> CheckIfRuleExistsAndReplaceInFile(SourceText result)
```

Comment:
Instead of writing this as `Tuple<SourceText, TextLine?, TextLine?>`, you can write the tuple as `(SourceText newText, TextLine? lastValidHeaderSpanEnd, TextLine? lastValidSpecificHeaderSpanEnd>)`. This way you can even update the callsites to use tuple names for clarity.